sources/main.c

```diff
@@ -26,7 +26,10 @@ int	main(int argc, char **argv, char **envp)
 	load_history();
 	while (TRUE)
 	{
-		input = readline(get_colored_prompt());
+		if (isatty(STDIN_FILENO))
+			input = readline(get_colored_prompt());
+		else
+			input = readline(NO_INTERACTIVE_PROMPT_TEXT);
```

Comment:
Por algum motivo ainda está ocorrendo o processamento do conteúdo de entrada processado pelo pipe e não apenas o resultado da execução entregue para o minishell.

```bash
jonnathan-ls@Dell-G15:/mnt/c/Users/jonnathan-ls/42-cursus/minishell$ echo pwd | ./minishell 
pwd
/mnt/c/Users/jonnathan-ls/42-cursus/minishell
jonnathan-ls@Dell-G15:/mnt/c/Users/jonnathan-ls/42-cursus/minishell$ echo "exit 15" | ./minishell
exit 15
jonnathan-ls@Dell-G15:/mnt/c/Users/jonnathan-ls/42-cursus/minishell$ echo $?
15
jonnathan-ls@Dell-G15:/mnt/c/Users/jonnathan-ls/42-cursus/minishell$ echo "echo OK" | ./minishell 
echo OK
OK
```

Reply:
O problema vem do processamento da `readline`

Quando ela não é processada não ocorre log do conteúdo fornecido pelo pipe. Então, foi necessário implementar separação para quando for mode não interativo, processar o conteudo sem a `readline` https://github.com/peda-cos/minishell/pull/52/commits/bb247d6f65c298bd02f8a10285e6b8fa41bce64a